datafusion/common/src/utils.rs

```diff
@@ -390,6 +391,27 @@ pub fn arrays_into_list_array(
     ))
 }
 
+/// Get the base type of a data type.
+///
+/// Example
+/// ```
+/// use arrow::datatypes::{DataType, Field};
+/// use datafusion_common::utils::base_type;
+/// use std::sync::Arc;
+///
+/// let data_type = DataType::List(Arc::new(Field::new("item", DataType::Int32, true)));
+/// let base_type = base_type(&data_type);
+/// assert_eq!(base_type, DataType::Int32);
+pub fn base_type(data_type: &DataType) -> DataType {
```

Comment:
```suggestion
/// ```
pub fn base_type(data_type: &DataType) -> DataType {
```